recipes/scip/all/conanfile.py

```diff
@@ -0,0 +1,154 @@
+        return {
+            "gcc": "5",
+            "clang": "4",
+            "apple-clang": "7",
+        }
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
+        if self.options.shared and self.options.with_sym == "bliss":
+            raise ConanInvalidConfiguration("Bliss is not supported in shared mode.")
+        comp = self.settings.compiler
+        if self.options.with_sym == "bliss" and comp == 'clang' and comp.libcxx and comp.libcxx == 'libc++':
+            raise ConanInvalidConfiguration("Bliss does not support libc++.")
+        if self.dependencies["soplex"].options.with_boost and not self.options.with_boost:
+            raise ConanInvalidConfiguration("The options 'with_boost' should be aligned with 'soplex:with_boost' too.")
+        if self.dependencies["soplex"].options.with_gmp and not self.options.with_gmp:
+            raise ConanInvalidConfiguration("The options 'with_gmp' should be aligned with 'soplex:with_gmp' too.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_boost:
+            self.requires("boost/1.81.0")
+        if self.options.with_gmp:
+            self.requires("gmp/6.2.1")
+        if self.options.with_sym == "bliss":
+            self.requires("bliss/0.77")
+        self.requires("soplex/6.0.3")
+        self.requires("zlib/1.2.13")
+
+    def configure(self):
+        self.options["soplex"].with_boost = self.options.with_boost
+        self.options["soplex"].with_gmp = self.options.with_gmp
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    @staticmethod
+    def _to_cmake(*arrays):
+        return ";".join(item.replace("\\", "/") for sublist in arrays for item in sublist)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["SHARED"] = self.options.shared
+        tc.variables["READLINE"] = False  # required for interactive stuff
+        tc.variables["GMP"] = self.options.with_gmp
+        tc.variables["BOOST"] = self.options.with_boost
```

Comment:
I didn't find BOOST variable in 8.0.3, how do you activate/deactivate it?

```
 % grep BOOST -r 
./INSTALL.md:| `BOOST_ROOT`           | `<custom/path/to/.../boost>`       | --                         | hint for location of boost                                         |
./src/cppad/core/testvector.hpp:$code CPPAD_BOOSTVECTOR$$ will be true.
./src/cppad/core/testvector.hpp:# if CPPAD_BOOSTVECTOR
./src/cppad/core/test_vector.hpp:of $code CPPAD_BOOSTVECTOR$$:
./src/cppad/core/test_vector.hpp:# if CPPAD_BOOSTVECTOR
./src/cppad/core/test_vector.hpp:# undef  CPPAD_BOOSTVECTOR
./src/cppad/core/test_vector.hpp:# define CPPAD_BOOSTVECTOR 0
./src/cppad/core/test_vector.hpp:$code CPPAD_BOOSTVECTOR$$ is true
./src/cppad/core/test_vector.hpp:# if CPPAD_BOOSTVECTOR
./src/cppad/core/undef.hpp:# undef CPPAD_BOOSTVECTOR
./src/cppad/configure.hpp:\def CPPAD_BOOSTVECTOR
./src/cppad/configure.hpp:# define CPPAD_BOOSTVECTOR 0
```

It's clear that Soplex needs boost, but it seems like libscip does not use it. Could you show where boost is used and activated?

Reply:
Indeed, just an include path depends on this setting, not the package itself. I dropped the option.